core/src/main/scala/stainless/extraction/imperative/AntiAliasing.scala

```
@@ -490,9 +496,11 @@ class AntiAliasing(override val s: Trees)(override val t: s.type)(using override
             if (targets.exists(target => !env.bindings.contains(target.receiver.toVal)))
               throw MalformedStainlessCode(up, "Unsupported form of array update")
 
+            val recV = transform(v, env)
+
             Block(targets map { target =>
-              val applied = updatedTarget(target, v)
-              transform(Assignment(target.receiver, applied).copiedFrom(up), env)
+              val applied = updatedTarget(target, recV)
+              Assignment(target.receiver, applied).copiedFrom(up)
```

I think this transformation (and a few others) might be incorrectly dropping stuff from the input array expression. If there was a `Block` expression in `a`, then I believe it would be silently dropped in the output here.

I'm not sure what the cleanest approach to fix this is, though. I guess some options are:
1. adding a method which extracts the non-target part of an expression and join it into the output,
2. we could always include the expression in the output and hope simplification cleans it up later on by having the result be something like
    ```scala
    Block(transform(a, env), Assignement(...))
    ```
3. we could have a pre-transformation which makes sure to bring all blocks/targets into a normal form which is compatible with target assignment. This option would also make sure that mutating calls inside targets are handled correctly.

---

Option 3 seems to also benefit https://github.com/epfl-lara/stainless/issues/1068